lib/service/storage/elasticsearch.ts

```diff
@@ -3218,9 +3436,48 @@ class ElasticSearch extends Service {
       searchBody.query = { match_all: {} };
     }
 
+    if (index && this.virtualIndex.isVirtual(index)) {
+      this._sanitizeSearchBodyForVirtualIndex(searchBody, index);
+    }
+
     return searchBody;
   }
 
+  _sanitizeSearchBodyForVirtualIndex(searchBody, index) {
+    const filteredQuery = {
+      bool: {
+        filter: [],
+        must: {},
+        should: null,
+      },
+    };
+    if (searchBody.query.bool) {
+      filteredQuery.bool = searchBody.query.bool;
```

Comment:
You should make a deep copy with `JSON.parse(JSON.stringify())` here to avoid mutate the original query

Reply:
Done